### PR TITLE
Add support for slowroll to bootloader_uefi

### PR DESCRIPTION
**Pull request by asdil12**
VR: https://openqa.opensuse.org/tests/3923808
Ticket: https://progress.opensuse.org/issues/135731

### CODE REVIEW

File: tests/jeos/firstrun.pm [hunk 1]
```diff
@@ -167,7 +167,7 @@ sub run {
     send_key 'ret';
 
     # Accept EULA if required
-    unless (is_tumbleweed || is_microos || is_leap('>=15.4')) {
+    unless (is_tumbleweed || is_slowroll || is_microos || is_leap('>=15.4')) {
         assert_screen 'jeos-doyouaccept';
         send_key 'ret';
     }
```
Comment (okurz) [inline tests/jeos/firstrun.pm:173]: Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

```suggestion
    if (is_leap('<15.4')) {
        assert_screen 'jeos-doyouaccept';
        send_key 'ret';
    }
```

and as Leap < 15.4 is EOL just delete the complete if-block

File: lib/bootloader_setup.pm [hunk 2]
```diff
@@ -383,7 +383,7 @@ sub uefi_bootmenu_params {
     assert_screen("grub2-enter-edit-mode", 30) if is_jeos;
     # Kiwi in TW uses grub2-mkconfig instead of the custom kiwi config
     # Locate gfxpayload parameter and update it
-    if (is_jeos && (is_tumbleweed || is_sle('>=15-sp1') || is_leap('>=15.1'))) {
+    if (is_jeos && (is_tumbleweed || is_slowroll || is_sle('>=15-sp1') || is_leap('>=15.1'))) {
```
Comment (okurz) [inline lib/bootloader_setup.pm:386]: Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

```suggestion
    unless (!is_sle('<15-sp1') || is_leap('<15.1')) {
```

but that would need testing if this works.

Comment (asdil12) [reply lib/bootloader_setup.pm:386]: I am pretty sure that this line does a completely different thing. But the `unless(!` is so hard to think through that I'm not sure.

This here *migth* be doing the same:
```perl
if (is_jeos && ((!is_sle('<15-sp1') && !is_leap('<15.1')) || check_var('DISTRI', 'opensuse'))) {
```

Comment (okurz) [reply lib/bootloader_setup.pm]: yes, sorry `unless (!…` does not make sense. Your version looks good. And, no, `if tumbleweed` as a version check should be avoided

File: tests/jeos/firstrun.pm [hunk 3]
```diff
@@ -167,7 +167,7 @@ sub run {
     send_key 'ret';
 
     # Accept EULA if required
-    unless (is_tumbleweed || is_microos || is_leap('>=15.4')) {
+    if (is_sle) {
```
Comment (mloviska) [inline tests/jeos/firstrun.pm:170]: License is shown in sle-micro as well

File: lib/bootloader_setup.pm [hunk 4]
```diff
@@ -383,7 +383,8 @@ sub uefi_bootmenu_params {
     assert_screen("grub2-enter-edit-mode", 30) if is_jeos;
     # Kiwi in TW uses grub2-mkconfig instead of the custom kiwi config
     # Locate gfxpayload parameter and update it
-    if (is_jeos && (is_tumbleweed || is_sle('>=15-sp1') || is_leap('>=15.1'))) {
+    if (is_jeos && ((!is_sle('<15-sp1') && !is_leap('<15.1')) || check_var('DISTRI', 'opensuse'))) {
```
Comment (mloviska) [inline lib/bootloader_setup.pm:386]: The oldest supported jeos based on sle is 12-sp5, hence it can be `is_jeos && !is_sle('=12-SP5') || is_opensuse`

File: lib/bootloader_setup.pm [hunk 5]
```diff
@@ -15,7 +15,7 @@ use Time::HiRes 'sleep';
 use testapi;
 use Utils::Architectures;
 use utils;
-use version_utils qw(is_microos is_sle_micro is_jeos is_leap is_sle is_tumbleweed is_selfinstall is_alp is_transactional);
+use version_utils qw(is_microos is_sle_micro is_jeos is_leap is_sle is_tumbleweed is_slowroll is_selfinstall is_alp is_transactional);
```
Comment (okurz) [inline lib/bootloader_setup.pm:18]: Now `is_tumbleweed` and `is_slowroll` are not used in this file anymore at all which is great. Please remove both includes. This also discourages other users to incorrectly use those methods.

File: lib/bootloader_setup.pm [hunk 6]
```diff
@@ -383,7 +383,8 @@ sub uefi_bootmenu_params {
     assert_screen("grub2-enter-edit-mode", 30) if is_jeos;
     # Kiwi in TW uses grub2-mkconfig instead of the custom kiwi config
     # Locate gfxpayload parameter and update it
-    if (is_jeos && (is_tumbleweed || is_sle('>=15-sp1') || is_leap('>=15.1'))) {
+    if (is_jeos && (!is_sle('=12-SP5') || is_opensuse)) {
+
```
Comment (okurz) [inline lib/bootloader_setup.pm:387]: unrelated change

File: tests/jeos/firstrun.pm [hunk 7]
```diff
@@ -13,7 +13,7 @@ use base "opensusebasetest";
 use strict;
 use warnings;
 use testapi;
-use version_utils qw(is_jeos is_sle is_tumbleweed is_leap is_opensuse is_microos is_sle_micro is_vmware is_alp is_bootloader_sdboot);
+use version_utils qw(is_jeos is_sle is_tumbleweed is_slowroll is_leap is_opensuse is_microos is_sle_micro is_vmware is_alp is_bootloader_sdboot);
```
Comment (okurz) [inline tests/jeos/firstrun.pm:16]: Same as above please remove the unused imports

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/bootloader_setup.pm (hunk 6 above)
Comment (asdil12) [reply lib/bootloader_setup.pm:387]: No it isn't.
This is an improvement on https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18622#pullrequestreview-1870323749

Comment (okurz) [reply lib/bootloader_setup.pm:387]: I was only relating to the blank line

**Review by okurz — APPROVED** [review]
